app/models/key-list.js

```diff
@@ -26,6 +26,14 @@ var KeyList = CachedList.extend({
     cluster: DS.attr(),
 
     /**
+     * The max amount of records that should be fetched in a given request
+     * @property maxKeysPerRequest
+     * @type Integer
+     * @default 100
+     */
+    maxKeysPerRequest: DS.attr('integer', { defaultValue: 100 }),
```

Comment:
This should be put on inherited cached list model. Will update with rest of feedback.